utils/paramfetch/src/lib.rs

```diff
@@ -48,18 +48,22 @@ struct ParameterData {
 }
 
 #[inline]
-fn param_dir() -> String {
-    std::env::var(DIR_ENV).unwrap_or_else(|_| PARAM_DIR.to_owned())
+fn param_dir(mut data_dir: PathBuf) -> String {
+    std::env::var(PathBuf::from(DIR_ENV)).unwrap_or_else(|_| {
+        data_dir.push(PARAM_DIR);
+        data_dir.to_str().unwrap().to_string()
+    })
```

Comment:
How about returning a `PathBuf` rather than a `String`?

Reply:
Indeed. So we either convert a `String` to a `PathBuf` or a `PathBuf` to a `String`. Which conversion can we do without the possibility of a panic?